### PR TITLE
POI parsing and spawning

### PR DESCRIPTION
**Pull request by SloopyNoSleep**
Code is a bit icky, more time needs to be spent on conceptualizing how to handle mapping between a simple "encompassing" type that gets deserialized from json and how it has logic that spawns appropriate gameobject with transform, relationships, and components.

Also I can't solve everything with inheritance like the current typeconverter leans into. Makes composition of objects hard especially without a 1-to-1 mapping to internal types. I'll probably make more commits to try to clean this up a little (but I can make those in a future PR if we don't mind the technical debt 😝 )

### CODE REVIEW

File: Winch/Serialization/DredgeTypeHelpers.cs [hunk 1]
```diff
@@ -15,20 +15,31 @@ public static class DredgeTypeHelpers
...
-        var jsonDict = JsonConvert.DeserializeObject<Dictionary<string, int>>(value.ToString());
+        var jsonDict = JsonConvert.DeserializeObject<Dictionary<string, int>>(value.ToString()) ?? throw new InvalidOperationException("Unable to parse color.");
         return GetColorFromJsonDictionary(jsonDict);
     }
 
     private static Color GetColorFromJsonDictionary(Dictionary<string, int> color)
     {
         int r = 0, g = 0, b = 0, a = 255;
-        if (color.TryGetValue("r", out var value)) r = value;
+
```
Comment (bdlm-dev) [inline Winch/Serialization/DredgeTypeHelpers.cs]: Red missing here

File: Winch/Serialization/Item/SpatialItemDataConverter.cs [hunk 2]
```diff
@@ -26,7 +26,7 @@ public class SpatialItemDataConverter : ItemDataConverter
         { "ignoreDamageWhenPlacing", new(false, null)},
         { "isUnderlayItem", new(false, null)},
         { "forbidStorageTray", new(false, null)},
-        { "dimensions", new(new List<Vector2Int>(){new Vector2Int(1,1)}, o => DredgeTypeHelpers.ParseDimensions((JArray)o)) },
+        { "dimensions", new(new List<Vector2Int>(){new(1,1)}, o => DredgeTypeHelpers.ParseDimensions((JArray)o)) },
```
Comment (bdlm-dev) [inline Winch/Serialization/Item/SpatialItemDataConverter.cs]: I believe this should be (0, 0)

File: Winch/Serialization/DredgeTypeHelpers.cs (hunk 1 above)
Comment (SloopyNoSleep) [reply Winch/Serialization/DredgeTypeHelpers.cs]: Darn, that's what I get for copy-pasting. Nice catch, thanks!

File: Winch/Serialization/Item/SpatialItemDataConverter.cs (hunk 2 above)
Comment (SloopyNoSleep) [reply Winch/Serialization/Item/SpatialItemDataConverter.cs]: Yes, you are very much right. Thank you

**Comment by Ashiepaws** [comment]
LGTM for now, we can always adjust type conversion later, this is a solid base for new modded content :)

**Pull request merged**